src/lib/routes/admin-api/user/user.ts

```diff
@@ -167,10 +168,14 @@ class UserController extends Controller {
         res: Response,
     ): Promise<void> {
         const { user } = req;
-        const { password, confirmPassword } = req.body;
-        if (password === confirmPassword) {
+        const { password, confirmPassword, oldPassword } = req.body;
+        if (password === confirmPassword && oldPassword != null) {
```

Comment:
Maybe it doesn't hurt to check for truthy here instead of `!= null` so we can cover `undefined` as well. But I'm guessing `undefined` is converted to `null` anyways.

```suggestion
        if (password === confirmPassword && oldPassword) {
```

Reply:
This part was test-driven and missing oldPassword is turned into null as you mentioned so I'd rather stick to the original version since I can't even trigger a test scenario for undefined to test it.